In [1]:
import pandas as pd
pd.set_option('display.max_columns',None)
import warnings
warnings.filterwarnings('ignore')

In [2]:
df = pd.read_csv(r'C:\Users\HP\Desktop\Python\Vs_Python\Data analysis\RCA-Sales-Super-Store\Dataset\Sample - Superstore.csv', encoding='latin1')

In [3]:
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,State,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,Kentucky,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,Kentucky,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,California,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


In [4]:
df.shape

(9994, 21)

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

In [6]:
df.isnull().sum()

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64

In [7]:
df.duplicated().sum()

np.int64(0)

In [8]:
df.columns

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit'],
      dtype='object')

In [9]:
orders = df[['Order ID', 'Order Date', 'Ship Date', 'Ship Mode']]
customers = df[['Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State','Postal Code', 'Region']]
products = df[['Product ID', 'Category', 'Sub-Category', 'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit','Order ID']]

In [10]:
total_sales = df['Sales'].sum()

In [11]:
total_sales

np.float64(2297200.8603000003)

In [12]:
total_profit = df['Profit'].sum()

In [13]:
total_profit

np.float64(286397.0217)

In [14]:
profit_margin = (total_profit / total_sales) * 100

print(f"Profit Margin: {profit_margin:.2f}%")

Profit Margin: 12.47%


In [15]:
df['Profit_Margin_%'] = (df['Profit'] / df['Sales']) * 100

In [16]:
products['Profit_Margin_%'] = (products['Profit'] / products['Sales']) * 100

### INSIGHT
    - The business generated ₹22.97 lakh in revenue with a profit margin of 12.47%, yielding approximately ₹2.86 lakh in profit.

In [17]:
products.groupby(['Category'])['Profit'].sum()

Category
Furniture           18451.2728
Office Supplies    122490.8008
Technology         145454.9481
Name: Profit, dtype: float64

- Furniture sale generate toal profit $18451.27 (Lowest)
- Office Supplies sale generate toal profit $122490.80 (Medium)
- Technology sale generate toal profit $145454.94 (Most)

In [18]:
products.groupby(['Category'])['Profit'].mean()

Category
Furniture           8.699327
Office Supplies    20.327050
Technology         78.752002
Name: Profit, dtype: float64

- Each Furniture sale generates about $8.70 profit
- Each Office Supplies sale generates about $20.33 profit
- Each Technology sale generates about $78.75 profit

In [19]:
products.groupby(['Category'])['Profit_Margin_%'].mean()

Category
Furniture           3.878353
Office Supplies    13.803029
Technology         15.613805
Name: Profit_Margin_%, dtype: float64

### INSIGHT
    - Technology Category generate 15% profit margin(Most Profitable Category)
    - Furniture Category generate 3% profit margin(Lowest profit Category)

In [20]:
products[products['Category'] == 'Furniture'].groupby('Sub-Category')['Profit'].sum().sort_values()

Sub-Category
Tables        -17725.4811
Bookcases      -3472.5560
Furnishings    13059.1436
Chairs         26590.1663
Name: Profit, dtype: float64

- In Furniture the Sub-category [Tables, Bookcases] are in Negative Profit

In [21]:
products[products['Category'] == 'Furniture'].groupby('Sub-Category')['Profit_Margin_%'].mean().sort_values()

Sub-Category
Tables        -14.772653
Bookcases     -12.664007
Chairs          4.389963
Furnishings    13.706635
Name: Profit_Margin_%, dtype: float64

- Tables is 14% Negative
- Bookcases is 12% Negative

In [23]:
products[products['Sub-Category'] == 'Tables'].groupby('Discount')[['Sales', 'Profit']].sum()

,Sales,Profit
Discount,,
0.00,71578.760,13276.2997
0.20,45430.232,-303.5580
0.30,25182.150,-3402.3276
0.40,45614.406,-16187.3968
0.45,5484.974,-2493.1111
0.50,13675.010,-8615.3873


- As Discount Increase Profit Decrease

In [24]:
df[df['Sub-Category'] == 'Tables'].groupby('State')['Profit'].sum().sort_values().head(10)

State
New York         -4535.6408
Illinois         -4309.7447
North Carolina   -3684.2539
Ohio             -2715.3345
Tennessee        -2663.4138
Pennsylvania     -2588.7538
Florida          -2493.1111
Arizona          -2281.8701
Texas            -2216.6766
Oregon           -1048.8971
Name: Profit, dtype: float64

- New York is the largest loss-maker

In [25]:
products[products['Sub-Category'] == 'Tables'].groupby('Product Name')['Profit'].sum().sort_values().head(10)

Product Name
Chromcraft Bull-Nose Wood Oval Conference Tables & Bases                       -2876.1156
Bush Advantage Collection Racetrack Conference Table                           -1934.3976
Balt Solid Wood Round Tables                                                   -1201.0581
BoxOffice By Design Rectangular and Half-Moon Meeting Room Tables              -1148.4375
Riverside Furniture Oval Coffee Table, Oval End Table, End Table with Drawer   -1147.4000
Bretford “Just In Time” Height-Adjustable Multi-Task Work Tables                -964.1940
Bevis Oval Conference Table, Walnut                                             -856.0144
BPI Conference Tables                                                           -795.9725
Hon 94000 Series Round Tables                                                   -681.2140
Chromcraft Bull-Nose Wood 48" x 96" Rectangular Conference Tables               -611.5878
Name: Profit, dtype: float64

In [26]:
df[df['Sub-Category'] == 'Tables'].groupby(['State', 'Discount'])['Profit'].sum().sort_values()

State           Discount
New York        0.40       -4535.6408
Illinois        0.50       -4309.7447
North Carolina  0.40       -3684.2539
Ohio            0.40       -2715.3345
Tennessee       0.40       -2663.4138
Pennsylvania    0.40       -2588.7538
Florida         0.45       -2493.1111
Arizona         0.50       -2281.8701
Texas           0.30       -2216.6766
Oregon          0.50       -1048.8971
Colorado        0.50        -974.8754
Massachusetts   0.30        -545.2947
California      0.20        -303.5580
Rhode Island    0.30        -230.1678
New Hampshire   0.30        -105.3164
Delaware        0.30         -85.8963
West Virginia   0.30         -76.9536
Maryland        0.30         -71.1170
New Jersey      0.30         -51.2906
Connecticut     0.30         -19.6146
Arkansas        0.00          45.2940
Iowa            0.00         106.6248
Georgia         0.00         137.9470
Wisconsin       0.00         147.5022
Kentucky        0.00         160.1586
Minnesota       0.00     

As Discount increase the profit descrease. Top state with Negative Profit has the Highest discount.

In [27]:
df[df['Sub-Category'] == 'Tables'] \
.groupby('Discount') \
.agg({
    'Sales':'sum',
    'Profit':'sum',
    'Order ID':'count'
})

,Sales,Profit,Order ID
Discount,,,
0.00,71578.760,13276.2997,72
0.20,45430.232,-303.5580,71
0.30,25182.150,-3402.3276,54
0.40,45614.406,-16187.3968,75
0.45,5484.974,-2493.1111,11
0.50,13675.010,-8615.3873,36


### Root Cause
    - Furniture losses are mainly caused by the Tables sub-category.
    - Tables become unprofitable once discounts exceed roughly 20%.
    - Discounts between 30%-50% generate severe losses.
    - States such as New York, Illinois, North Carolina, Ohio, and Tennessee are major contributors because many table sales there occur at these high discount levels.
    - Tables sold without discounts achieve an 18.55% profit margin, proving the product itself is profitable.

In [37]:
df.columns

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit',
       'Profit_Margin_%', 'duration'],
      dtype='object')

In [44]:
df.groupby(['Customer Name'])['Sales'].sum().sort_values(ascending=False).head(5)

Customer Name
Sean Miller      25043.050
Tamara Chand     19052.218
Raymond Buch     15117.339
Tom Ashbrook     14595.620
Adrian Barton    14473.571
Name: Sales, dtype: float64

- Sean Miller brings the most money to the business

In [78]:
df.groupby(['Customer Name'])['Sales'].mean().sort_values(ascending=False).head(5)

Customer Name
Mitch Willingham    1751.292000
Sean Miller         1669.536667
Tamara Chand        1587.684833
Grant Thornton      1558.535333
Tom Ashbrook        1459.562000
Name: Sales, dtype: float64

- Mitch Willingham spends the most each time they purchase

In [45]:
df.groupby(['Customer Name'])['Profit'].sum().sort_values(ascending=False).head(5)

Customer Name
Tamara Chand     8981.3239
Raymond Buch     6976.0959
Sanjit Chand     5757.4119
Hunter Lopez     5622.4292
Adrian Barton    5444.8055
Name: Profit, dtype: float64

In [81]:
df[df['Customer Name'] == 'Tamara Chand']

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,State,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit,Profit_Margin_%,duration
2072,2073,CA-2016-129693,2016-11-26,2016-12-02,Standard Class,TC-20980,Tamara Chand,Corporate,United States,Seattle,Washington,98105,West,OFF-BI-10002954,Office Supplies,Binders,Newell 3-Hole Punched Plastic Slotted Magazine...,7.312,2,0.2,2.5592,35.00,6
3185,3186,CA-2014-123498,2014-11-07,2014-11-09,First Class,TC-20980,Tamara Chand,Corporate,United States,Houston,Texas,77041,Central,OFF-BI-10000632,Office Supplies,Binders,Satellite Sectional Post Binders,26.046,3,0.8,-44.2782,-170.00,2
3186,3187,CA-2014-123498,2014-11-07,2014-11-09,First Class,TC-20980,Tamara Chand,Corporate,United States,Houston,Texas,77041,Central,OFF-EN-10004773,Office Supplies,Envelopes,Staple envelope,74.352,3,0.2,26.9526,36.25,2
6825,6826,CA-2016-118689,2016-10-02,2016-10-09,Standard Class,TC-20980,Tamara Chand,Corporate,United States,Lafayette,Indiana,47905,Central,OFF-ST-10001558,Office Supplies,Storage,Acco Perma 4000 Stacking Storage Drawers,32.480,2,0.0,4.8720,15.00,7
6826,6827,CA-2016-118689,2016-10-02,2016-10-09,Standard Class,TC-20980,Tamara Chand,Corporate,United States,Lafayette,Indiana,47905,Central,TEC-CO-10004722,Technology,Copiers,Canon imageCLASS 2200 Advanced Copier,17499.950,5,0.0,8399.9760,48.00,7
6827,6828,CA-2016-118689,2016-10-02,2016-10-09,Standard Class,TC-20980,Tamara Chand,Corporate,United States,Lafayette,Indiana,47905,Central,OFF-BI-10004600,Office Supplies,Binders,Ibico Ibimaster 300 Manual Binding System,735.980,2,0.0,331.1910,45.00,7
6828,6829,CA-2016-118689,2016-10-02,2016-10-09,Standard Class,TC-20980,Tamara Chand,Corporate,United States,Lafayette,Indiana,47905,Central,OFF-BI-10003712,Office Supplies,Binders,"Acco Pressboard Covers with Storage Hooks, 14 ...",34.370,7,0.0,16.8413,49.00,7
6829,6830,CA-2016-118689,2016-10-02,2016-10-09,Standard Class,TC-20980,Tamara Chand,Corporate,United States,Lafayette,Indiana,47905,Central,OFF-AR-10001958,Office Supplies,Art,Stanley Bostitch Contemporary Electric Pencil ...,33.960,2,0.0,9.5088,28.00,7
8060,8061,CA-2015-156013,2015-09-20,2015-09-26,Standard Class,TC-20980,Tamara Chand,Corporate,United States,Long Beach,New York,11561,East,OFF-PA-10001970,Office Supplies,Paper,Xerox 1881,61.400,5,0.0,28.8580,47.00,6
8061,8062,CA-2015-156013,2015-09-20,2015-09-26,Standard Class,TC-20980,Tamara Chand,Corporate,United States,Long Beach,New York,11561,East,OFF-BI-10002437,Office Supplies,Binders,Recycled Premium Regency Composition Covers,24.448,2,0.2,8.8624,36.25,6


In [46]:
df.groupby(['Customer Name'])['Discount'].mean().sort_values(ascending=False).head(5)

Customer Name
Thais Sissman       0.700000
Ritsa Hightower     0.533333
Vivek Sundaresam    0.475000
Gene Hale           0.466667
Lisa Hazard         0.466667
Name: Discount, dtype: float64

In [47]:
df[df['Customer Name'] == 'Thais Sissman']

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,State,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit,Profit_Margin_%,duration
1450,1451,US-2015-160150,2015-07-19,2015-07-20,First Class,TS-21085,Thais Sissman,Consumer,United States,Phoenix,Arizona,85023,West,OFF-BI-10004352,Office Supplies,Binders,Wilson Jones DublLock D-Ring Binders,2.025,1,0.7,-1.3500,-66.666667,1
2120,2121,US-2017-168690,2017-01-07,2017-01-13,Standard Class,TS-21085,Thais Sissman,Consumer,United States,Ormond Beach,Florida,32174,South,OFF-BI-10000145,Office Supplies,Binders,Zipper Ring Binder Pockets,2.808,3,0.7,-1.9656,-70.000000,6


In [49]:
df[df['Customer Name'] == 'Ritsa Hightower']

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,State,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit,Profit_Margin_%,duration
2748,2749,CA-2015-110247,2015-12-04,2015-12-08,Standard Class,RH-19555,Ritsa Hightower,Consumer,United States,Tallahassee,Florida,32303,South,OFF-BI-10001553,Office Supplies,Binders,SpineVue Locking Slant-D Ring Binders by Cardinal,8.226,3,0.7,-6.0324,-73.333333,4
8648,8649,CA-2016-164574,2016-07-29,2016-08-03,Standard Class,RH-19555,Ritsa Hightower,Consumer,United States,Charlotte,North Carolina,28205,South,OFF-ST-10003816,Office Supplies,Storage,Fellowes High-Stak Drawer Files,704.760,5,0.2,26.4285,3.750000,5
8649,8650,CA-2016-164574,2016-07-29,2016-08-03,Standard Class,RH-19555,Ritsa Hightower,Consumer,United States,Charlotte,North Carolina,28205,South,OFF-BI-10002429,Office Supplies,Binders,"Premier Elliptical Ring Binder, Black",27.396,3,0.7,-20.0904,-73.333333,5


In [28]:
products.groupby('Category')['Profit'].sum().sort_values()

Category
Furniture           18451.2728
Office Supplies    122490.8008
Technology         145454.9481
Name: Profit, dtype: float64

In [29]:
products.groupby('Category')['Sales'].sum().sort_values()

Category
Office Supplies    719047.0320
Furniture          741999.7953
Technology         836154.0330
Name: Sales, dtype: float64

In [30]:
df['Order Date'] = pd.to_datetime(df['Order Date'])

In [31]:
df['Ship Date'] = pd.to_datetime(df['Ship Date'])

In [32]:
df['duration'] = (df['Ship Date'] - df['Order Date']).dt.days

In [33]:
df['Ship Mode'].value_counts().sort_values()

Ship Mode
Same Day           543
First Class       1538
Second Class      1945
Standard Class    5968
Name: count, dtype: int64

In [34]:
df.groupby('Ship Mode')['Profit_Margin_%'].mean().sort_values()

Ship Mode
Standard Class    11.007329
First Class       11.594781
Same Day          13.815135
Second Class      15.020878
Name: Profit_Margin_%, dtype: float64

In [35]:
df.groupby('Ship Mode')['Profit'].sum().sort_values()

Ship Mode
Same Day           15891.7589
First Class        48969.8399
Second Class       57446.6354
Standard Class    164088.7875
Name: Profit, dtype: float64

In [36]:
df.columns

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit',
       'Profit_Margin_%', 'duration'],
      dtype='object')